src/Sentry/Protocol/Envelopes/EnvelopeItem.cs

```diff
@@ -200,6 +201,20 @@ public static EnvelopeItem FromTransaction(Transaction transaction)
         return new EnvelopeItem(header, new JsonSerializable(transaction));
     }
 
+    /// <summary>
+    /// Creates an <see cref="EnvelopeItem"/> from <paramref name="profileInfo"/>.
+    /// </summary>
+    internal static EnvelopeItem FromProfileInfo(Task<ProfileInfo?> profileInfo)
+    {
+        var header = new Dictionary<string, object?>(1, StringComparer.Ordinal)
+        {
+            [TypeKey] = TypeValueProfile
+        };
+
+        var jsonSerializable = profileInfo.ContinueWith(task => task.Result as IJsonSerializable);
+        return new EnvelopeItem(header, new AsyncJsonSerializable(jsonSerializable));
```

Comment:
This is a task, right?

```suggestion
        var jsonSerializableTask = profileInfo.ContinueWith(task => task.Result as IJsonSerializable);
        return new EnvelopeItem(header, new AsyncJsonSerializable(jsonSerializableTask));
```